--- ktorrent-2.2/libktorrent/net/downloadthread.cpp.orig	2007-07-02 12:48:46.000000000 -0400
+++ ktorrent-2.2/libktorrent/net/downloadthread.cpp	2007-07-05 04:01:34.187340212 -0400
@@ -18,7 +18,6 @@
  *   51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.          *
  ***************************************************************************/
 #include <math.h>
-#include <sys/poll.h>
 #include <util/functions.h>
 #include "socketgroup.h"
 #include "downloadthread.h"
--- ktorrent-2.2/libktorrent/net/downloadthread.h.orig	2007-07-02 12:48:46.000000000 -0400
+++ ktorrent-2.2/libktorrent/net/downloadthread.h	2007-07-05 04:01:29.829819844 -0400
@@ -21,10 +21,9 @@
 #define NETDOWNLOADTHREAD_H
 
 #include <vector>
+#include <sys/poll.h>
 #include "networkthread.h"
 
-struct pollfd;
-
 namespace net
 {
 
--- ktorrent-2.2/libktorrent/torrent/authenticationmonitor.cpp.orig	2007-07-02 12:48:47.000000000 -0400
+++ ktorrent-2.2/libktorrent/torrent/authenticationmonitor.cpp	2007-07-05 01:48:21.385508086 -0400
@@ -19,7 +19,6 @@
  ***************************************************************************/
 #include <math.h>
 #include <unistd.h>
-#include <sys/poll.h>
 #include <util/functions.h>
 #include <util/log.h>
 #include <mse/streamsocket.h>
--- ktorrent-2.2/libktorrent/torrent/authenticationmonitor.h.orig	2007-07-02 12:48:47.000000000 -0400
+++ ktorrent-2.2/libktorrent/torrent/authenticationmonitor.h	2007-07-05 01:52:15.326096549 -0400
@@ -22,8 +22,7 @@
 
 #include <list>
 #include <vector>
-		
-struct pollfd;
+#include <sys/poll.h>
 
 namespace bt
 {
--- ktorrent-2.2/plugins/infowidget/flagdb.cpp.orig	2007-07-02 12:48:43.000000000 -0400
+++ ktorrent-2.2/plugins/infowidget/flagdb.cpp	2007-07-05 04:51:46.705315995 -0400
@@ -37,6 +37,13 @@
 {
 }
 
+kt::FlagDBSource& kt::FlagDBSource::operator = ( kt::FlagDBSource const& rhs )
+{
+	type = rhs.type;
+	pathPattern = rhs.pathPattern;
+	return *this;
+}
+
 QString kt::FlagDBSource::FlagDBSource::getPath(const QString& country) const
 {
 	if (type) {
--- ktorrent-2.2/plugins/infowidget/flagdb.h.orig	2007-07-02 12:48:43.000000000 -0400
+++ ktorrent-2.2/plugins/infowidget/flagdb.h	2007-07-05 04:54:29.724112084 -0400
@@ -32,14 +32,15 @@
 		FlagDBSource();
 		FlagDBSource(const char* type, const QString& pathPattern);
 		FlagDBSource(const QString& pathPattern);
+		FlagDBSource& operator = ( FlagDBSource const& );
 		QString getPath(const QString& country) const;
 
-		const char* getType() { return type; };
-		const QString& getPathPattern() { return pathPattern; };
+		char const* getType() const { return type; };
+		QString const& getPathPattern() const { return pathPattern; };
 
 	private:
-		const char* type;
-		const QString pathPattern;
+		char const* type;
+		QString pathPattern;
 	};
 
 	/**
